xdsl/dialects/riscv.py

```diff
@@ -876,6 +877,19 @@ class JalOp(RdImmOperation):
 
     name = "riscv.jal"
 
+    def __init__(
+        self,
+        immediate: int | AnyIntegerAttr | str | LabelAttr,
+        *,
+        rd: RegisterType | Register | None = None,
+        comment: str | StringAttr | None = None,
+    ):
+        if rd is None:
+            # By default place return address in
+            rd = RegisterType(Registers.RA)
```

Comment:
I'm thinking that we should address this cleanly, not sure whether we should do it before or after merging this

Reply:
Moved RA assignment from JalOp to CallOp in this [commit](https://github.com/xdslproject/xdsl/pull/929/commits/b7cc12d6dd62b7a8dcef756bd9a06aec61ebb1b2)